src/SwiftUI/SwiftUI/Color.cs

```diff
@@ -0,0 +1,226 @@
+﻿using System;
+using System.Buffers;
+using System.Runtime.InteropServices;
+
+using Swift.Interop;
+
+namespace SwiftUI
+{
+	using static Color;
+
+	public enum RGBColorSpace
+	{
+		DisplayP3,
+		RGB,
+		RGBLinear,
+	};
+
+	[SwiftImport (SwiftUILib.Path)]
+	public unsafe partial class Color : View
+	{
+		public IntPtr Data { get; private set; }
+
+		#region Static Colour Spaces
+		public static IntPtr ColorSpaceDisplayP3 {
+			get {
+				var fullMeta = RGBColorSpaceMetadata (0);
+				var meta = &fullMeta->Metadata;
+				return GetColorSpaceDisplayP3 (meta);
+			}
+		}
+
+		public static IntPtr ColorSpaceRGB => GetColorSpaceRGB ();
+		public static IntPtr ColorSpaceRGBLinear => GetColorSpaceRGBLinear ();
+		#endregion
+
+		#region Static Colours
+		static Color? black = null;
+		public static Color Black => black ??= new Color (GetColorBlack (0));
+
+		static Color? blue = null;
+		public static Color Blue => blue ??= new Color (GetColorBlue (0));
+
+		public static Color Clear => new Color (GetColorClear (0));
+
+		static Color? gray = null;
+		public static Color Gray => gray ??= new Color (GetColorGray (0));
+
+		static Color? green = null;
+		public static Color Green => green ??= new Color (GetColorGreen (0));
+
+		static Color? orange = null;
+		public static Color Orange => orange ??= new Color (GetColorOrange (0));
+
+		static Color? pink = null;
+		public static Color Pink => pink ??= new Color (GetColorPink (0));
+
+		public static Color Primary => new Color (GetColorPrimary (0));
+
+		static Color? purple = null;
+		public static Color Purple => purple ??= new Color (GetColorPurple (0));
+
+		static Color? red = null;
+		public static Color Red => red ??= new Color (GetColorRed (0));
+
+		public static Color Secondary => new Color (GetColorSecondary (0));
+
+		static Color? white = null;
+		public static Color White => white ??= new Color (GetColorWhite (0));
+
+		static Color? yellow = null;
+		public static Color Yellow => yellow ??= new Color (GetColorYellow (0));
+		#endregion
+
+		protected override void InitNativeData (void* handle)
+		{
+			// Using TransferFuncType.InitWithTake here in case we leak.
+			using (var dataHandle = Data.GetSwiftHandle ())
+				SwiftType.Transfer (handle, dataHandle.Pointer, TransferFuncType.InitWithTake);
+		}
+
+		#region Constructors
+		internal Color (IntPtr data)
+		{
+			Data = data;
+		}
+
+		public Color (RGBColorSpace colorSpace, double red, double green, double blue, double opacity)
+		{
+			Data = CreateFromRGBColorSpaceRedGreenBlueOpacity ((ulong)colorSpace, red, green, blue, opacity);
+		}
+
+		public Color (RGBColorSpace colorSpace, double white, double opacity)
+		{
+			Data = CreateFromRGBColorSpaceWhiteOpacity ((ulong)colorSpace, white, opacity);
```

Comment:
Same as previous comment

Reply:
As it looks like these will take a little time to workout, I've removed all the `RGBColorSpace` constructors for now, so a working version can go in. Other constructors have been test and work.